### PR TITLE
Admin metaboxes views translation escaping fix

### PR DESCRIPTION
**Pull request by khanrn**
### includes/admin/meta-boxes/views
- Data escaped and validated
- Translators text placeholder documentation fix. Like `/* translators: %s: download count */`.

### CODE REVIEW

File: includes/abstracts/abstract-wc-data.php [hunk 1]
```diff
@@ -398,7 +398,7 @@ public function add_meta_data( $key, $value, $unique = false ) {
 	 *
 	 * @param  string $key
 	 * @param  string $value
-	 * @param  int $meta_id
+	 * @param  string $meta_id
```
Comment (mikejolley) [inline includes/abstracts/abstract-wc-data.php]: Should be int.

File: includes/admin/meta-boxes/views/html-order-download-permission.php [hunk 2]
```diff
@@ -5,30 +5,48 @@
...
+						$product->get_name(),
+						$download->get_product_id(),
+						$download->get_order_id(),
+						$download->get_order_key(),
+						$download->get_download_id()
+					)
+				),
 				esc_html( $file_count ),
-				esc_html( wc_get_filename_from_url( $product->get_file_download_path( $download->get_download_id() ) ) )
+				esc_html(
```
Comment (mikejolley) [inline includes/admin/meta-boxes/views/html-order-download-permission.php]: I don't think this is necessary? Same for the above filter. Unless there is a PHPCS issue, it should be left.

File: includes/admin/meta-boxes/views/html-order-items.php [hunk 3]
```diff
@@ -155,23 +158,26 @@
...
 				<tr>
-					<td class="label"><?php echo $tax->label; ?>:</td>
+					<td class="label"><?php echo esc_html( $tax->label ); ?>:</td>
 					<td width="1%"></td>
-					<td class="total"><?php
-						if ( ( $refunded = $order->get_total_tax_refunded_by_rate_id( $tax->rate_id ) ) > 0 ) {
+					<td class="total">
+						<?php
+						$refunded = $order->get_total_tax_refunded_by_rate_id( $tax->rate_id );
+						if ( ( $refunded ) > 0 ) {
```
Comment (mikejolley) [inline includes/admin/meta-boxes/views/html-order-items.php]: Why extra brackets here?

File: includes/admin/meta-boxes/views/html-order-shipping.php [hunk 4]
```diff
@@ -83,17 +83,17 @@
 						<div class="view">
 							<?php
 								echo ( '' !== $tax_item_total ) ? wc_price( wc_round_tax_total( $tax_item_total ), array( 'currency' => $order->get_currency() ) ) : '&ndash;';
-
-								if ( $refunded = $order->get_tax_refunded_for_item( $item_id, $tax_item_id, 'shipping' ) ) {
+							$refunded = $order->get_tax_refunded_for_item( $item_id, $tax_item_id, 'shipping' );
```
Comment (mikejolley) [inline includes/admin/meta-boxes/views/html-order-shipping.php]: Indenting

File: includes/admin/meta-boxes/views/html-product-attribute.php [hunk 5]
```diff
@@ -14,24 +14,27 @@
...
-						<input type="hidden" name="attribute_position[<?php echo $i; ?>]" class="attribute_position" value="<?php echo esc_attr( $attribute->get_position() ); ?>" />
+						<input type="hidden" name="attribute_position[<?php echo esc_attr( $i ); ?>]" class="attribute_position" value="<?php echo esc_attr( $attribute->get_position() ); ?>" />
 					</td>
 					<td rowspan="3">
-						<label><?php _e( 'Value(s)', 'woocommerce' ); ?>:</label>
+						<label><?php esc_html_e( 'Value(s)', 'woocommerce' ); ?>:</label>
 
-						<?php if ( $attribute->is_taxonomy() && ( $attribute_taxonomy = $attribute->get_taxonomy_object() ) ) : ?>
+						<?php
+						$attribute_taxonomy = $attribute->get_taxonomy_object();
```
Comment (mikejolley) [inline includes/admin/meta-boxes/views/html-product-attribute.php]: This should be left as it was or tweaked; you're getting a taxonomy object before checking if it's a taxonomy.

File: includes/admin/meta-boxes/views/html-product-variation-download.php [hunk 6]
```diff
@@ -5,10 +5,10 @@
...
-	<td class="file_url"><input type="text" class="input_text" placeholder="<?php esc_attr_e( "http://", 'woocommerce' ); ?>" name="_wc_variation_file_urls[<?php echo $variation_id; ?>][]" value="<?php echo esc_attr( $file['file'] ); ?>" /></td>
-	<td class="file_url_choose" width="1%"><a href="#" class="button upload_file_button" data-choose="<?php esc_attr_e( 'Choose file', 'woocommerce' ); ?>" data-update="<?php esc_attr_e( 'Insert file URL', 'woocommerce' ); ?>"><?php echo str_replace( ' ', '&nbsp;', __( 'Choose file', 'woocommerce' ) ); ?></a></td>
-	<td width="1%"><a href="#" class="delete"><?php _e( 'Delete', 'woocommerce' ); ?></a></td>
+	<td class="file_url"><input type="text" class="input_text" placeholder="<?php esc_attr_e( 'http://', 'woocommerce' ); ?>" name="_wc_variation_file_urls[<?php echo esc_attr( $variation_id ); ?>][]" value="<?php echo esc_attr( $file['file'] ); ?>" /></td>
+	<td class="file_url_choose" width="1%"><a href="#" class="button upload_file_button" data-choose="<?php esc_attr_e( 'Choose file', 'woocommerce' ); ?>" data-update="<?php esc_attr_e( 'Insert file URL', 'woocommerce' ); ?>"><?php echo esc_html__( 'Choose file', 'woocommerce' ); ?></a></td>
```
Comment (mikejolley) [inline includes/admin/meta-boxes/views/html-product-variation-download.php]: esc_attr_e can replace esc_attr__

Comment (mikejolley) [reply includes/admin/meta-boxes/views/html-product-variation-download.php]: Also did you test what impact removing non breaking spaces had on ui?

**Review by mikejolley — CHANGES_REQUESTED** [review]

File: includes/abstracts/abstract-wc-data.php (hunk 1 above)
Comment (rnaby) [reply includes/abstracts/abstract-wc-data.php]: OK. Then we should change `$meta_id = ''` to `$meta_id = 0` at line *403*, cause empty string and 0 are same for `if` condition check.

File: includes/admin/meta-boxes/views/html-order-download-permission.php (hunk 2 above)
Comment (rnaby) [reply includes/admin/meta-boxes/views/html-order-download-permission.php]: Yes, it was detected by **PHPCS**.

File: includes/admin/meta-boxes/views/html-order-items.php (hunk 3 above)
Comment (rnaby) [reply includes/admin/meta-boxes/views/html-order-items.php]: Removed the extra bracket.

File: includes/admin/meta-boxes/views/html-product-variation-download.php (hunk 6 above)
Comment (rnaby) [reply includes/admin/meta-boxes/views/html-product-variation-download.php]: Yes. I tested the impact and it seemed OK to me.

File: includes/admin/meta-boxes/views/html-order-download-permission.php (hunk 2 above)
Comment (claudiosanches) [reply includes/admin/meta-boxes/views/html-order-download-permission.php]: Not using our `phpcs.ruleset.xml`.

Comment (mikejolley) [reply includes/admin/meta-boxes/views/html-order-download-permission.php]: Not on my install...these types of change are unnecessary. cc @claudiosanches

Comment (rnaby) [reply includes/admin/meta-boxes/views/html-order-download-permission.php]: I don't know which standard you are using, but I'm using *WordPress VIP Coding Standard*. @mikejolley and @claudiosanches

Comment (rnaby) [reply includes/admin/meta-boxes/views/html-order-download-permission.php]: I removed the escaping from both this and the above filter. @mikejolley

**Comment by mikejolley** [comment]
In your last commit you removed escapes but left the items spanning multiple lines :/ The escape was not the problem.

**Comment by rnaby** [comment]
@mikejolley Sorry for the inconvenience, it happened because of my coding habit. I usually try to put everything with in 80 columns. I fixed it now. Anyway, as WordPress VIP standard is catching those not escaping using `esc_html()` a warning, will you folks accept with escaping ? If yes then I'll add escaping and if not then I'll remove the escaping from the filter.

**Comment by mikejolley** [comment]
It had escaping before so that escaping should be left in place.

**Comment by rnaby** [comment]
@mikejolley I've pushed an update as instructed. Please do check when you get time.

File: includes/admin/meta-boxes/views/html-order-items.php (hunk 3 above)
Comment (mikejolley) [reply includes/admin/meta-boxes/views/html-order-items.php]: This is still `( $refunded )` - why are those there?

Comment (rnaby) [reply includes/admin/meta-boxes/views/html-order-items.php]: Please have a look [here].(https://github.com/woocommerce/woocommerce/pull/17776/commits/c909d1a4d34bd43ccc1c49901d9e811da997b629) At this commit I changed the bracket issue. Don't know why GitHub hasn't updated this.

Comment (mikejolley) [reply includes/admin/meta-boxes/views/html-order-items.php]: Maybe it's not pushed to this PR/branch?

Comment (rnaby) [reply includes/admin/meta-boxes/views/html-order-items.php]: No. It looks like it's pushed to the correct branch. Please have a look, it says `rnaby:admin-metaboxes-views-translation-escaping-fix` branch.

Comment (mikejolley) [reply includes/admin/meta-boxes/views/html-order-items.php]: Maybe it's cached. Ill check again.

Comment (mikejolley) [reply includes/admin/meta-boxes/views/html-order-items.php]: Definitely there. I just checked out the branch.

Comment (rnaby) [reply includes/admin/meta-boxes/views/html-order-items.php]: Is there anything I can do. Though, it says my local and GitHub branch is updated with that code.

Comment (mikejolley) [reply includes/admin/meta-boxes/views/html-order-items.php]: Commit it again?

Comment (rnaby) [reply includes/admin/meta-boxes/views/html-order-items.php]: Commit has been done. Please check again.

**Pull request merged**